### ISSUE TITLE
"testnames" can't find all the tests if suite had Multi-level parent-child relationship

### ISSUE THREAD

**Issue by eyehwan** [testng-team/testng#1634]
### TestNG Version
6.13
### Expected behavior
When  Multi-level parent-child relationship existed:
1. "testnames" should support to find test not only from first level child suite, but also deeper child suites.
2.  test names provided by ”testnames” can be from different suites. 

### Actual behavior
Which is not so far.

### Is the issue reproductible on runner?

All

**Comment by krmahadevan** [comment]
> "testnames" should support to find test not only from first level child suite, but also deeper child suites.

I dont think TestNG supported nesting of suites beyond one level. So you would have a parent suite, which contains one or more child suites. That's it. I dont remember us going beyond that.

Lets ignore the support for `filter by testnames`. Have you seen TestNG work with just a `suite of suite of suites` ? I would be curious to know the results of that.

ping @cbeust @juherr -Please advise if this is something that TestNG should be supporting ?

**Comment by eyehwan** [comment]
`I dont think TestNG supported nesting of suites beyond one level. `
If you can confirmed that TestNG never support multi-level parent-child relationship for "testnames", then my guess is right, this is a new feature request. 

From the descriptions of "testnames" from TestNG documentation is:
`Only tests defined in a <test> tag matching one of these names will be run.`
But in current solution, it only support to fetcht test names from first level child suite at most, which make users confused. Because if it supports nest child test name fetching, why only first level, not all nested levels?

`Have you seen TestNG work with just a suite of suite of suites ? I would be curious to know the results of that.`
Definately for sure, nested hierarchy is very nice feature and widely used by TestNG, not only test names, but also make tests structure and logic more flexible. 

@testng-team, again if you think it is a new feature request around testnames, I am very appreciated that any comments from you. I hope to make it happen.

**Comment by krmahadevan** [comment]
@eyehwan,

> But in current solution, it only support to fetcht test names from first level child suite at most, which make users confused

I dont think there was any confusion to begin with, atleast not to the best of my knowledge from looking at the questions on the `testng-users` google forums in the past couple of years.

Let me rephrase my question. Currently does `suite of suite of suites` work for you ? If it doesn't then we need to address it as a requirement in total. That would basically need us to call out what all should be applicable at the outer level and what all should be applicable at the inner level.

**Comment by juherr** [comment]
@krmahadevan Do you remember why only one level is supported?

**Comment by krmahadevan** [comment]
@juherr - Nope. Not sure.

**Comment by eyehwan** [comment]
`Let me rephrase my question. Currently does suite of suite of suites work for you ?`
No, it doesn't as supposed to be. Because we used to fetch first level test names before, but recently when we want to fetch deeper level test names, it turn out not.

**Comment by eyehwan** [comment]
`That would basically need us to call out what all should be applicable at the outer level and what all should be applicable at the inner level.`
@krmahadevan , @juherr  Can you tell me what should be addressed here? Because it looks like we need discussion to make get finalized decision. 
But in this case, if first level nested already supported, why not deeper also. Pandora box already open.  :)
